packages/base/src/types.ts

```diff
@@ -40,3 +42,58 @@ declare global {
     jupytergisMaps: { [name: string]: Map };
   }
 }
+
+/**
+ * Color ramp types and definitions
+ */
+export type ColorRampType = 'Sequential' | 'Divergent' | 'Cyclic';
+
+export interface IBaseColorRampDefinition {
+  type: ColorRampType;
+}
+
+export interface ISequentialColorRampDefinition
+  extends IBaseColorRampDefinition {
+  type: 'Sequential';
+}
+
+export interface IDivergentColorRampDefinition
+  extends IBaseColorRampDefinition {
+  type: 'Divergent';
+  criticalValue: number;
+}
+
+export interface ICyclicColorRampDefinition extends IBaseColorRampDefinition {
+  type: 'Cyclic';
+}
+
+export type IColorRampDefinition =
+  | ISequentialColorRampDefinition
+  | IDivergentColorRampDefinition
+  | ICyclicColorRampDefinition;
+
+export interface IColorMap {
+  name: ColorRampName;
+  colors: string[];
+  definition: IColorRampDefinition;
+  userMin?: number;
+  userMax?: number;
+  criticalValue?: number;
+}
+
+export type ColorRampName = keyof typeof COLOR_RAMP_DEFINITIONS;
+
+export interface IColorRampValueControlsProps {
```

Comment:
Since this will only be used by one component, I don't think we should include it here.